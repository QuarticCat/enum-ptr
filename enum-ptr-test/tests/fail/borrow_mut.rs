#![allow(dead_code)]

use enum_ptr::{EnumPtr, Unit};

#[derive(EnumPtr)]
#[enum_ptr(borrow_mut)]
#[repr(C, usize)]
enum Foo<'a> {
    A(&'a mut i32),
    B(Unit),
}

#[derive(EnumPtr)]
#[enum_ptr(borrow_mut)]
#[repr(C, usize)]
enum Bar<'a> {
    A(&'a mut i32),
    #[enum_ptr(skip)]
    B(Unit),
}

#[derive(EnumPtr)]
#[enum_ptr(borrow_mut(rename = "RefMutBaz"))]
#[repr(C, usize)]
enum Baz<'a> {
    A(&'a mut i32),
    #[enum_ptr(skip_borrow_mut)]
    B(Unit),
}

fn main() {}
