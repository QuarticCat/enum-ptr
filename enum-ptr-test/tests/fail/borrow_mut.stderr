error[E0277]: the trait bound `Unit: FieldDerefMut` is not satisfied
 --> tests/fail/borrow_mut.rs:5:10
  |
5 | #[derive(EnumPtr)]
  |          ^^^^^^^ the trait `FieldDerefMut` is not implemented for `Unit`
  |
  = help: the following other types implement trait `FieldDerefMut`:
            &mut T
            Box<T>
            Option<&mut T>
            Option<Box<T>>
  = note: this error originates in the derive macro `EnumPtr` (in Nightly builds, run with -Z macro-backtrace for more info)
